core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -262,35 +277,33 @@ private static DataSource createRealmDataSource(RealmConfiguration realmConfig)
                     }
                 }
             }
-            rs = prepStmt.executeQuery();
-            List<String> lst = new ArrayList<String>();
-            while (rs.next()) {
-                String name = rs.getString(1);
-                lst.add(name);
-            }
-            if (lst.size() > 0) {
-                values = lst.toArray(new String[lst.size()]);
+            try(ResultSet rs = prepStmt.executeQuery() ) {
+                List<String> lst = new ArrayList<String>();
+                while (rs.next()) {
+                    String name = rs.getString(1);
+                    lst.add(name);
+                }
+                if (lst.size() > 0) {
+                    values = lst.toArray(new String[lst.size()]);
+                }
+                return values;
             }
```

Comment:
resultset is not in scope outside the code block, so it isn't available to call or pass into a close statement.  Tried putting in a catch statement for the sqlrecoverable exception but that threw an exception on missing return values.